src/modules/listclass.js

```diff
@@ -0,0 +1,50 @@
+export default class TodoList {
+    allTodos = [];
+
+    saveTodo() {
+      const todos = JSON.stringify(this.allTodos);
+      localStorage.setItem('todos', todos);
+    }
+
+    addTodo(description) {
+      const todo = {
+        description,
+        completed: false,
+        index: this.allTodos.length + 1,
+      };
+      this.allTodos.push(todo);
+      this.saveTodo();
+    }
+
+    deleteTodo(indx) {
+      this.allTodos.splice(indx - 1, 1);
+      this.allTodos.forEach((t) => {
+        if (t.index > indx) {
+          t.index -= 1;
+        }
+      });
+      this.saveTodo();
+    }
+
+    getStoredTodos() {
+      this.allTodos = JSON.parse(localStorage.getItem('todos'));
+    }
+
+    editTodo(newValue, index) {
+      this.allTodos[index - 1].description = newValue;
+      this.saveTodo();
+    }
+
+    completedTodo(status, index) {
+      this.allTodos[index - 1].completed = status;
+      this.saveTodo();
+    }
+
+    clearCompleted() {
+      this.allTodos = this.allTodos.filter((b) => b.completed === false);
+      for (let i = 0; i < this.allTodos.length; i += 1) {
+        this.allTodos[i].index = i + 1;
+      }
```

Comment:
- [ ] The logic is perfect. However, I think it will be more readable to chain a `.map` method right after the `.filter.

Suggestion:
you can use the example bellow

> ` this.allTodos = this.allTodos.filter((b) => b.completed === false)
                                                 .map((b, index) => {
                                                                               b.index = index + 1;
                                                                               return b;
                                                                               });`

Reply:
Thank you @Sboursen, I have done the suggested changes!